### ISSUE TITLE
[Bug]: Issue when regenerating a key via the UI

### ISSUE THREAD

**Issue by ncecere** [BerriAI/litellm#6350]
### What happened?

When trying to regenerate a key via the UI I get the following error

![image](https://github.com/user-attachments/assets/eb569a42-3fe9-4a2e-8d34-6760c62a7de9)

I would expect that a user could regenerate a key when they want via the UI.

### Relevant log output

```shell
{"message": "litellm.proxy.proxy_server.user_api_key_auth(): Exception occured - This key is made for LiteLLM UI, Tried to access route: /key/598b12e2a6233e4bc7328ec2a202e0d2b375d614ea8c5e49594cdb67cce8e7a5/regenerate. Not allowed\nRequester IP Address:10.51.18.7", "level": "ERROR", "timestamp": "2024-10-21T14:36:04.968554"}
```

**Comment by ncecere** [comment]
I forgot to mention in the initial response we are running v1.50.1-stable

**Comment by ishaan-jaff** [comment]
fixed on #6430 @ncecere
